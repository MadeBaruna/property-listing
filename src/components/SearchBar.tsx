import React from 'react';
import styled from 'styled-components';
import MdiSearchIcon from 'mdi-react/SearchIcon';

const Container = styled.div<{ focused: boolean }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  background: white;
  height: 40px;
  margin-left: 20px;
  box-shadow: ${({ focused }) => focused ? `0 0 0 2px #04ACF4` : `0 0 0 2px rgba(0, 0, 0, .1)`};
  max-width: 500px;
  width: 100%;
  border-radius: 2px;
`;

const Input = styled.input`
  display: flex;
  height: 100%;
  border: none;
  padding: 0 10px;
  flex: 1;
  font-size: 16px;
  width: 100%;
  background: transparent;

  &:focus {
    outline: none;
  }
`;

const Select = styled.select`
  margin-right: 5px;
  height: 100%;
  border: none;
  font-size: 16px;

  &:focus {
    outline: none;
  }
`;

const SearchIconButton = styled.button<{ focused: boolean }>`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${({ focused }) => focused ? '#04ACF4' : 'rgba(0, 0, 0, .1)'};
  width: 50px;
  height: 100%;
  border: none;
  cursor: pointer;

  &:focus {
    outline: none;
  }
`;

const SearchIcon = styled(MdiSearchIcon)<{ focused: boolean }>`
  fill: ${({ focused }) => focused ? 'white' : 'black'};
`;

const SearchBar: React.FC = () => {
  const [focused, setFocused] = React.useState(false);

  return (
    <Container focused={focused}>
      <Input onFocus={() => setFocused(true)} onBlur={() => setFocused(false)} />
      <Select onFocus={() => setFocused(true)} onBlur={() => setFocused(false)}>
        <option>Apartment</option>
        <option>Office</option>
      </Select>
      <SearchIconButton focused={focused}>
        <SearchIcon focused={focused} size={30} />
      </SearchIconButton>
    </Container>
  );
}

export default SearchBar;
