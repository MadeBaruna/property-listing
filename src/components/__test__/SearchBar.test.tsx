import React from 'react';
import 'jest-styled-components'
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router';

import SearchBar from '../SearchBar';

it('renders <SearchBar />', () => {
  const wrapper = render(
    <MemoryRouter>
      <SearchBar />
    </MemoryRouter>
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders <SearchBar /> with initial query', () => {
  const wrapper = render(
    <MemoryRouter initialEntries={['/?search=apartment&type=OFFICE']}>
      <SearchBar />
    </MemoryRouter>
  );

  expect(wrapper).toMatchSnapshot();
});
