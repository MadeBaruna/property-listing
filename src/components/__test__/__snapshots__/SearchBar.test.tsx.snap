// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <SearchBar /> 1`] = `
.c0 {
  background: white;
  margin-left: 20px;
  box-shadow: 0 0 0 2px rgba(0,0,0,.1);
  max-width: 500px;
  width: 100%;
  border-radius: 2px;
}

.c0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: none;
  padding: 0 10px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  width: 100%;
  background: transparent;
}

.c1:focus {
  outline: none;
}

.c2 {
  margin-right: 5px;
  height: 100%;
  border: none;
  font-size: 16px;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,.1);
  width: 50px;
  height: 100%;
  border: none;
  cursor: pointer;
}

.c3:focus {
  outline: none;
}

.c4 {
  fill: black;
}

.c4.focused {
  fill: white;
}

<div
  class="c0"
>
  <form>
    <input
      class="c1"
      value=""
    />
    <select
      class="c2"
    >
      <option
        selected=""
        value="APARTMENT"
      >
        Apartment
      </option>
      <option
        value="OFFICE"
      >
        Office
      </option>
    </select>
    <button
      class="c3"
    >
      <svg
        class="mdi-icon c4"
        fill="currentColor"
        height="30"
        viewBox="0 0 24 24"
        width="30"
      >
        <path
          d="M9.5,3C13.09,3 16,5.91 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16C5.91,16 3,13.09 3,9.5C3,5.91 5.91,3 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
        />
      </svg>
    </button>
  </form>
</div>
`;

exports[`renders <SearchBar /> with initial query 1`] = `
.c0 {
  background: white;
  margin-left: 20px;
  box-shadow: 0 0 0 2px rgba(0,0,0,.1);
  max-width: 500px;
  width: 100%;
  border-radius: 2px;
}

.c0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: none;
  padding: 0 10px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  width: 100%;
  background: transparent;
}

.c1:focus {
  outline: none;
}

.c2 {
  margin-right: 5px;
  height: 100%;
  border: none;
  font-size: 16px;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,.1);
  width: 50px;
  height: 100%;
  border: none;
  cursor: pointer;
}

.c3:focus {
  outline: none;
}

.c4 {
  fill: black;
}

.c4.focused {
  fill: white;
}

<div
  class="c0"
>
  <form>
    <input
      class="c1"
      value=""
    />
    <select
      class="c2"
    >
      <option
        selected=""
        value="APARTMENT"
      >
        Apartment
      </option>
      <option
        value="OFFICE"
      >
        Office
      </option>
    </select>
    <button
      class="c3"
    >
      <svg
        class="mdi-icon c4"
        fill="currentColor"
        height="30"
        viewBox="0 0 24 24"
        width="30"
      >
        <path
          d="M9.5,3C13.09,3 16,5.91 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16C5.91,16 3,13.09 3,9.5C3,5.91 5.91,3 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
        />
      </svg>
    </button>
  </form>
</div>
`;
